### PR TITLE
RBAC rules generated by Bazel 

### LINKED ISSUE #89: Utilize bazel for CRD and RBAC generation
Follow the code here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/543

/help
/milestone baseline

### PR DESCRIPTION
**Pull request by soggiest**
**What this PR does / why we need it**:
This PR brings us inline with the work being done on the AWS cloud provider repo. This PR generates RBAC at the time of build, which was missing prior.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #89

**Release note**:

```release-note
Adds Bazel code to generate necessary RBAC rules during a build.
```

### CODE REVIEW

File: config/default/rbac_role_binding_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: manager-rolebinding
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: azure-provider-system
```
Comment (tariq1890) [inline config/default/rbac_role_binding_patch.yaml]: Should this be done as well ?
```suggestion
    namespace: system
```

Comment (soggiest) [reply config/default/rbac_role_binding_patch.yaml]: Yes it should, good catch

Comment (soggiest) [reply config/default/rbac_role_binding_patch.yaml]: Weird, in my editor it shows that as having been changed.

**Comment by soggiest** [comment]
Closing this to switch out the fork. I'll be opening a new PR that incorporates the changes described above.

**Pull request closed without merging**